src/key-handler/index.tsx

```diff
@@ -21,29 +20,15 @@ type KeyboardEventHandler = (event: KeyboardEvent) => void;
 
 export interface KeyHandlerProps {
   triggers: Trigger[];
-  handler: KeyboardEventHandler;
-}
-
-function createMatchingKeyHandler(
-  triggers: Trigger[],
-  handler: KeyboardEventHandler
-): KeyboardEventHandler {
-  return (event: KeyboardEvent): void => {
-    if (triggers.length === 0) {
-      return;
-    }
-
-    if (triggers.some(matchTrigger.bind(null, event))) {
-      handler(event);
-    }
-  };
+  handler?: KeyboardEventHandler;
+  children?: ReactNode;
```

Comment:
Instead of both of these being optional, can we make the props a union type? I know this makes the destructuring a little harder, but otherwise `<KeyHandler triggers={[]} />` is a valid component but has no clear valid meaning in our case. And also a keyhandler with both children and a handler.

Something like:

```ts
interface LeafProp {
  handler: KeyboardEventHandler;
}

interface NodeProp {
  children: ReactNode;
}

type KeyHandlerProps = { triggers: Trigger[] } & (LeafProp | NodeProp);
```

This might be tricky to get right when trying to use the `handler` property on the props object since TypeScript is a little picky about accessing properties on a union object like this without a discriminant.

Another option would be to have two components, one for leaves and one for nodes, but that's a pretty big refactor so I hope we can get something working with the union type.